distribution/client/src/lib/snapshotter.js

```diff
@@ -74,6 +74,9 @@ class Snapshotter {
     this.git('commit', '--allow-empty', '--message', `${message}`);
   }
 
+  revertToLevelBefore(currentLevel) {
+    logger.error(`[NOT IMPLEMENTED YET]${LOG_PREFIX} Revert UL-${currentLevel} to previous Update Level state`);
```

Comment:
To be written, not the most complex part though. See the UL revert below for the more complex IMO.